coreos-release-note-generator/release-note-generator.py

```diff
@@ -0,0 +1,167 @@
+
+This file contains the following functions:
+
+    * parse_args - parse command line arguments
+    * read_yaml_snippets - read yaml snippets from `release-notes.d` directory
+    * write_yaml_snippets - write yaml snippets to STDOUT or `release-notes.yaml` or `release-notes.json`
+    * main - the main function of the script
+"""
+
+
+import yaml
+import argparse
+import os
+import glob
+import json
+import sys
+
+
+def parse_args():
+    """
+    Returns:
+        Namspace: Parsed command line attributes
+    """
+    parser = argparse.ArgumentParser(
+        description="Builds 'release-notes.yaml' from yaml snippets \
+            under 'release-notes.d' directory with specified build id. \
+            Outputs to STDOUT by default.")
+
+    parser.add_argument(
+        '--build-id', help='build id for release notes', required=True)
+    parser.add_argument(
+        '--config-dir', help="FCOS config directory where 'release-notes.d' lives", required=True)
+    parser.add_argument(
+        '--release-notes-file', help="input 'release-notes.yaml' for update, omit to generate a new one", required=False)
+    parser.add_argument(
+        '--output-dir', help="output directory for 'release-notes.yaml'", required=False)
+    parser.add_argument('--json', action='store_true',
+                        help='output json instead of yaml', required=False)
+    args = parser.parse_args()
+    return args
+
+
+def read_yaml_snippets(args):
+    """Reads and parses yaml snippets under `release-notes.d` directory
+
+    Args:
+        args (Namespace): Parsed command line attributes
+
+    Returns:
+        dictionary: A dictionary consists of a new release note item generated from
+                    yaml snippets under `release-notes.yaml`. As an example:
+                    {"ignition": [{subject: "", body: ""}]}
+    """
+    if not os.path.exists(args.config_dir):
+        raise Exception(
+            "config directory '{}' does not exist".format(args.config_dir))
+
+    if not os.path.exists(os.path.join(args.config_dir, 'release-notes.d')):
+        raise Exception(
+            "release-notes.d does not exist under {}".format(args.config_dir))
+
+    snippet_yaml_list = glob.glob(os.path.join(
+        args.config_dir, 'release-notes.d/*.yaml'))
+    if len(snippet_yaml_list) == 0:
+        print("release-notes.d/ does not contain any yaml snippets under '{}'".format(args.config_dir))
+        exit(0)
+
+    snippet_dict = dict()
+    for snippet_yaml in snippet_yaml_list:
+        with open(snippet_yaml, 'r') as f:
+            snippet = yaml.load(f, Loader=yaml.FullLoader)
+            for item in snippet:
+                note = {'subject': item.get(
+                    'subject', ''), 'body': item.get('body', '')}
+                # purposely avoid item.get('component', '') to error out if the component key does not exist
+                project_name = item['component']
```

Comment:
instead of using `project` throughout this file should we just call it `component` instead?

Reply:
Agreed :+1: